test/Sources/SalesforceSourceTests.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Streams.Dsl;
+using Arcane.Framework.Sinks.Parquet.Models;
+using Arcane.Framework.Sources.SalesForce;
+using Arcane.Framework.Sources.SalesForce.Services.AuthenticatedMessageProviders;
+using Arcane.Framework.Tests.Fixtures;
+using Moq;
+using Xunit;
+
+namespace Arcane.Framework.Tests.Sources;
+
+public class SalesforceSourceTests : IClassFixture<AkkaFixture>
+{
+    private readonly AkkaFixture akkaFixture;
+
+    private readonly SalesForceJobProvider jobProvider;
+    private readonly Mock<HttpClient> mockHttp;
+
+
+    public SalesforceSourceTests(AkkaFixture akkaFixture)
+    {
+        this.akkaFixture = akkaFixture;
+        this.mockHttp = new Mock<HttpClient>();
+        this.jobProvider = new SalesForceJobProvider("test.my.salesforce.com", "client_id", "client_secret", "user_name", "password", "security_token", "v60.0", 5);
+
+    }
+
+    [Fact]
+    public async Task TokenGeneration()
```

Comment:
It's usually better to distinguish the different steps of a test (like `Arrange`, `Act` and `Assert`) with comments  as described here: https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

You can see an example in [Arcane.Operator](https://github.com/SneaksAndData/arcane-operator/blob/80dbb86410c49f01c1607e36dc51d335b5c4a260/test/Services/StreamClassOperatorServiceTests.cs#L74)